rewrite-toml/src/main/java/org/openrewrite/toml/ChangeTableRowValue.java

```diff
@@ -0,0 +1,157 @@
+import org.intellij.lang.annotations.Language;
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.toml.tree.Toml;
+import org.openrewrite.toml.tree.TomlKey;
+import org.openrewrite.toml.tree.TomlValue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class ChangeTableRowValue extends Recipe {
+    @Option(displayName = "Table name",
+            description = "The name of the TOML array table containing the row to update.",
+            example = "package.contributors")
+    String tableName;
+
+    @Option(displayName = "Identifying key",
+            description = "The key within a table row to match on.",
+            example = "name")
+    String identifyingKey;
+
+    @Option(displayName = "Identifying value",
+            description = "The value to match. Can be a regular expression if useRegex is true.",
+            example = "Alice Smith")
+    String identifyingValue;
+
+    @Option(displayName = "Use regex",
+            description = "Whether to interpret the identifying value as a regular expression. Default is false.",
+            required = false)
+    @Nullable
+    Boolean useRegex;
+
+    @Option(displayName = "Property key",
+            description = "The key of the property to update within the matched row.",
+            example = "email")
+    String propertyKey;
+
+    @Option(displayName = "New value",
+            description = "The new value to set for the property. If null, the property will be removed.",
+            example = "\"alice.new@example.com\"",
+            required = false)
+    @Nullable
+    String newValue;
+
+    @Override
+    public String getDisplayName() {
+        return "Change TOML table row value";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Change a value in a TOML table row when the identifying property matches the specified matcher.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new TomlVisitor<ExecutionContext>() {
+            @Override
+            public Toml visitTable(Toml.Table table, ExecutionContext ctx) {
+                // Check if this is the correct table
+                if (table.getName() == null || !tableName.equals(table.getName().getName())) {
+                    return super.visitTable(table, ctx);
+                }
+
+                // Check if this table row matches the identifying criteria
+                if (!TableRowMatcher.hasMatchingKeyValue(table, identifyingKey, identifyingValue, useRegex)) {
+                    return super.visitTable(table, ctx);
+                }
+
+                // Update the property value
+                List<Toml> updatedValues = new ArrayList<>(table.getValues());
+                boolean hasChanges = false;
```

Comment:
This type of pattern makes me wonder if we could/should be using `ListUtils`. Was that something you've looked at?